packages/client/src/internals/types.test.ts

```diff
@@ -29,6 +31,19 @@ describe('AbortController', () => {
 });
```

Comment:
On line 23, under `getAbortController`, we have this lovely failure:

```plaintext
Argument of type 'typeof AbortController' is not assignable to parameter of type 'Maybe<AbortControllerEsque>'.
  Construct signature return types 'AbortController' and 'AbortControllerInstanceEsque' are incompatible.
    The types of 'signal' are incompatible between these types.
      Type 'AbortSignal' is missing the following properties from type 'AbortSignal': reason, throwIfAbortedts
```

Upstream issue for `abort-controller` not implementing `reason`: https://github.com/mysticatea/abort-controller/issues/36

But, per https://developer.mozilla.org/en-US/docs/Web/API/AbortController#browser_compatibility, Node has supported `AbortController` natively since v15.0.0. Node 14 is going EOL in a few months according to https://endoflife.date/nodejs. Does tRPC still want to support the `abort-controller` polyfill?

Reply:
> My maybe-naive outsider-perspective suggestion would be to remove this specific test line now, and then remove abort-controller from the samples in the next tRPC major version. 🤷

I have now played with this myself and I think this is a good course of action. We can remove all examples with `abort-controller` and simply not support it _now_, we don't need to wait for a new major

Let's do it